### ISSUE TITLE
Production of dev.bundle fails in ear with multiple wars

### ISSUE THREAD

**Issue by guttormvik2** [vaadin/flow#20358]
### Describe the bug

I'm upgrading from 24.5.0.beta5 to 24.5.1, so dev.bundle needs to be recreated.
At startup I get this:

>  Failed to find the following css files in the `node_modules` or `C:\DeveloperArea\dawinci-main\vaadin-upgrade\ptsmc-widgetset\src\main\frontend` directory tree:
      - leaflet/dist/leaflet.css

While the actual file looks like this:
> leaflet\dist\leaflet.css.DELETE.462c023d5a95cd0bbb12b41499d3a5eb

It looks like the cause is that DevModeInitializer is running in parallel, once for each of my wars.

### Reproduction

My setup is a multi-module maven project with a vaadin "widgetset" as one of the modules.
This produces an ear containing several wars and a vaadin "widgetset" jar, which we run on wildfly.

At boot, I got two of these:

> 14:44:50,550 ERROR [com.vaadin.flow.server.frontend.BundleUtils] (ServerService Thread Pool -- 119) Failed to copy existing `package-lock.json` to use: java.nio.file.FileSystemException: C:\DeveloperArea\dawinci-main\vaadin-upgrade\ptsmc-widgetset\target\dev-bundle\package-lock.json -> 
...
14:44:50,550 ERROR [com.vaadin.flow.server.frontend.BundleUtils] (ServerService Thread Pool -- 114) Failed to copy existing `package-lock.json` to use: java.nio.file.FileSystemException: C:\DeveloperArea\dawinci-main\vaadin-upgrade\ptsmc-widgetset\target\dev-bundle\package-lock.json -> 

And then later on, when I accessed the 1st vaadin page I got

>  Failed to find the following css files in the `node_modules` or `C:\DeveloperArea\dawinci-main\vaadin-upgrade\ptsmc-widgetset\src\main\frontend` directory tree:
      - leaflet/dist/leaflet.css

Checking in node_modules I saw that the file had been renamed for some reason:
> leaflet\dist\leaflet.css.DELETE.462c023d5a95cd0bbb12b41499d3a5eb

Suspecting that Vaadin is interfering with itself while processing things in parallel, I set a breakpoint in DevModeInitializer.initDevModeHandler

After doing a clean-frontend and a restart, the breakpoint was hit 4 times. I let the 1st one complete before letting the other ones continue, and now everything works as expected.

Looking at the context at each breakpoint, it looks like DevModeInitializer is called once for each war in the ear, no matter if the war uses vaadin or not.

### System Info

Windows 10 Enterprise 22H2
Firefox 131.0.3 (64-bit)
Vaadin 24.5.1

**Comment by tltv** [comment]
`DevModeStartupListener` is in Flow's `vaadin-dev-server` dependency, which is in Platform's `vaadin-dev` dependency. Just as `hilla-dev`.

You could try excluding whole `vaadin-dev` dependency from the shared libraries. And add it to only one WAR. It's basically same as with production packaging where you can exclude whole `vaadin-dev` when development mode dependencies are not needed.

**Comment by guttormvik2** [comment]
Also tried to exclude/include vaadin-dev-server instead of vaadin-dev. That gave the same ClassCastException as above.

Looks like I have to continue to rely on the same workaround as before: Set a breakpoint in Vaadin's code to prevent things running in parallel.

Or, maybe I should just drop dev-mode completely; So far it has had 0 value and a huge cost. However, I am kind of hoping that the AI stuff will have relevance at some point in the future

**Comment by mcollovati** [comment]
A potential cause could be that you have `flow-server` both in ear and in the war and `com.vaadin.flow.di.LookupInitializer$LookupImpl` and `com.vaadin.flow.di.Lookup` are loaded from different class loaders.

**Comment by guttormvik2** [comment]
@mcollovati, yes, it is probably a classloader issue, but I only have flow-server in ptsmc.ear\lib

To stake a step back, we are looking at various hacks to work around an issue now.
The real issue is that for some reason it seems Vaadin doesn't support:
* Classical ear application with multiple wars and multiple vaadin entrypoints
* Using skinnywar to not duplicate the libraries (We have 100MB in ear\lib, though the vaadin jars are only around 7MB)
* Using a shared widgetset/frontend (bundles are only 2M, but node_modules is 150M with 23K files!)

Ideally, this should be something that Vaadin supported. Currently I can achieve it with various hacks, but I'm never sure if the next vaadin version will break it.
